ruby/lib/minitest/queue/runner.rb

```diff
@@ -197,9 +199,18 @@ def report_grind_command
           abort! error.message
         end
 
-        reporter = GrindReporter.new(build: supervisor.build)
-        reporter.report
-        exit! reporter.success? ? 0 : 1
+        grind_reporter = GrindReporter.new(build: supervisor.build)
+        grind_reporter.report
+
+        test_time_record = CI::Queue::Redis::TestTimeRecord.new(queue_url, queue_config)
+        test_time_reporter = Minitest::Queue::TestTimeReporter.new(
+          build: test_time_record,
+          limit: queue_config.max_test_duration,
+          percentile: queue_config.max_test_duration_percentile,
+        )
+        test_time_reporter.report
+
+        exit! grind_reporter.success? && test_time_reporter.success? ? 0 : 1
```

Comment:
As we add another option to fail here I was thinking that it would be good for the grind reporter to give us some feedback on what failed that is parseable by a machine. This would give us the ability to collect metrics on how often things fail. Exporting the result as a json would be fine. I was thinking about an exit code but that would just hide more complex situations where multiple things fail.

Reply:
Yeah, I was thinking of something similar. Right now, a test takes too long, the message "Do you think you have intermittently failing tests?" pop up in shopify core.